src/api/app/controllers/labels_controller.rb

```diff
@@ -0,0 +1,58 @@
+class LabelsController < ApplicationController
+  skip_before_action :require_login, only: [:index]
+  before_action :xml_hash, :find_labelable_and_label_template, only: [:create]
+
+  # GET /labels/:project/:package
+  # GET /labels/:number
+  def index
+    if params[:number]
+      labelable = BsRequest.find_by_number(params[:number])
+      message = "Unable to find request '#{params[:number]}'"
+    else
+      labelable = Package.find_by_project_and_name(params[:project], params[:package])
+      message = "Unable to find project '#{params[:project]}' and package '#{params[:package]}'"
+    end
+
+    render_error(status: 404, message: message) && return unless labelable
+
+    @labels = labelable.labels
+  end
+
+  # POST /labels
+  def create
+    authorize @labelable, :update_labels?
+    authorize @label_template, :update?
+
+    label = Label.new(@parsed_xml)
+    if label.save
+      render_ok
+    else
+      render_error(status: 400, message: label.errors.full_messages.to_sentence)
+    end
+  end
+
+  private
+
+  def xml_hash
+    request_body = request.body.read
+    @parsed_xml = (Xmlhash.parse(request_body) || {}).with_indifferent_access if request_body.present?
```

Comment:
When this will be `nil`?

Reply:
When the request body is nil, we receive a 400 error. This can happen when a request is sent without any body. Or were you referring to something else?

Reply:
No, I was referring to that exactly, but you already check that `request_body.present?`, so it won't be `nil` to let the `{}` kick in, right?

Reply:
If the request_body is empty, then `Xmlhash.parse(request_body)` will return `nil`. Calling `with_indifferent_access` on `nil` will raise an error. Wrapping it with `{}` ensures that `with_indifferent_access` is called on an empty hash instead, preventing the error

Reply:
Ah right. `request_body.present?` is already taking care of that. Good catch. I've changed it